### PR TITLE
External Secrets Operator

### PR DESCRIPTION
**Pull request by hauskaffee**
Adding a how to doc for External Secrets Operator

### CODE REVIEW

**Comment by hauskaffee** [comment]
Closed as this is going to be a blog post

File: _docs/kb/articles/external-secrets-operators-aws-secrets-manager.md [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+- GitOps Runtime Installed
+- A Git Source is added to the Runtime
+  - Usage in this doc has it pointing to the path of `gitops/argocd`
+- AWS Account
+
+## Details
+
+### Installation
+
+First, you need to install the External Secrets Operator. To do that, we are going to add an Application to your Git Source. Create a file in your Git Source called `external-secrets-operator.yaml` and use the below application to install. Once done, save, commit, and push to your repo.
```
Comment (lrochette) [inline _docs/kb/articles/external-secrets-operators-aws-secrets-manager.md:29]: the below application manifest to install it

File: _docs/kb/articles/external-secrets-operators-aws-secrets-manager.md [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+---
+
+## Overview
+
+This article explains how to install and use the [External Secrets Operator](https://external-secrets.io/latest/) with [AWS Secrets Manager](https://aws.amazon.com/secrets-manager/).
+
+### Pre-Reqs
+
+- GitOps Runtime Installed
+- A Git Source is added to the Runtime
```
Comment (lrochette) [inline _docs/kb/articles/external-secrets-operators-aws-secrets-manager.md:21]: Maybe add a link for runtime and git source installation

File: _docs/kb/articles/external-secrets-operators-aws-secrets-manager.md [hunk 3]
```diff
@@ -0,0 +1,267 @@
...
+                "arn:aws:secretsmanager:us-east-1:<ACCOUNT_ID>>:secret:testing/*"
+            ]
+        }
+    ]
+}
+```
+
+{% endraw %}
+
+Once the IRSA is created, create a secret in AWS Secrets Manager (region us-east-1 in this example). Use the "Other type of Secret" when creating the secret. Add the key-value pairs that you want. When naming, use the prefix of `testing/` for this example. The rest of the options, use the defaults.
```
Comment (lrochette) [inline _docs/kb/articles/external-secrets-operators-aws-secrets-manager.md:99]: For the rest of the options

File: _docs/kb/articles/external-secrets-operators-aws-secrets-manager.md [hunk 4]
```diff
@@ -0,0 +1,267 @@
...
+```shell
+├── gitops
+│   ├── argocd
+│   │   └── external-secrets-operator.yaml
+│   ├── test-applications
+```
+
+{% endraw %}
+
+Inside test-applications directory create a file called `secret-store.yaml`.  Here we will create a Service Account and Secret Store config.  The SecretStore will allow us to access AWS Secrets Manager and use the Service Account to make the API Calls to AWS.
```
Comment (lrochette) [inline _docs/kb/articles/external-secrets-operators-aws-secrets-manager.md:116]: Inside the test-applications directory, create
There are 2 spaces before "Here"
and a Secret Store Config
Tee Secret Store

File: _docs/kb/articles/external-secrets-operators-aws-secrets-manager.md [hunk 5]
```diff
@@ -0,0 +1,267 @@
...
+
+```yaml
+apiVersion: external-secrets.io/v1beta1
+kind: SecretStore
+metadata:
+  name: secretstore-sample # the name you want to call the Secret Store
+spec:
+  provider:
+    aws:
+      service: SecretsManager # Specifing AWS Scret Manager
```
Comment (lrochette) [inline _docs/kb/articles/external-secrets-operators-aws-secrets-manager.md:128]: AWS Secret Manager ('e' missing in Secret)

File: _docs/kb/articles/external-secrets-operators-aws-secrets-manager.md [hunk 6]
```diff
@@ -0,0 +1,267 @@
...
+kind: ServiceAccount
+metadata:
+  annotations:
+    eks.amazonaws.com/role-arn: arn:aws:iam::<ACCOUNT_ID>:role/<ROLE_NAME> # The Role that you created to have access to Secrets Manager 
+  name: aws-secret-store
+```
+
+{% endraw %}
+
+Now create another file called `external-secret.yaml` in the testing-applications directory.  This is where we are going to use to generate a kubernets secret.  We will define a refresh interval so the screte is up to date in the cluster, how to access the secret via the Secret Store, the name of the scret in AWS Secret Manager, and what to name the k8s secret kind once retrieved.
```
Comment (lrochette) [inline _docs/kb/articles/external-secrets-operators-aws-secrets-manager.md:145]: This is where we are going to use to generate a kubernetes secret: this does not sounds right
Maybe "This is how we are going to generate a Kubernetes secret from the AWS one"
 so the "Secret" is up to date (Secret misspelled), and again later in "secret in AWS Secret Manager"

File: _docs/kb/articles/external-secrets-operators-aws-secrets-manager.md [hunk 7]
```diff
@@ -0,0 +1,267 @@
...
+
+{% raw %}
+
+```yaml
+apiVersion: external-secrets.io/v1beta1
+kind: ExternalSecret
+metadata:
+  name: example-secret
+spec:
+  refreshInterval: 1h # allows us to update secret if values change
```
Comment (lrochette) [inline _docs/kb/articles/external-secrets-operators-aws-secrets-manager.md:155]: I remember reading something about API limit or cost associated with the call. Maybe add a blurb around that so people don't put 5 seconds

File: _docs/kb/articles/external-secrets-operators-aws-secrets-manager.md (hunk 2 above)
Comment (lrochette) [reply _docs/kb/articles/external-secrets-operators-aws-secrets-manager.md:21]: nvm, I see the links at the bottom

**Pull request closed without merging**